content/knowledge/scalinghistory.md

```diff
@@ -0,0 +1,68 @@
+   PoW by itself isn’t very new. Original similar implementation was thought of in 1992 to [combat email spam](https://en.wikipedia.org/wiki/Hashcash). So every email will have some form of simple proof of work so that it is easy for me to send email to a friend but gets very hard for a spammer to send millions of emails at the same time.
+   In theory, PoW can scale infinitely. But with it comes the extra overhead of computational complexity which may not be practical in a truly global sense. Also some PoW based blockchains like Ethereum1.0 suffer from huge transaction fees as well as low throughput of 10-15 transactions per second which make them impractical for true global usage.
+
+## 2. Federated Byzantine Agreement (FBA)
+
+   After Bitcoin and PoW came into picture in 2008, another class of consensus protocols surfaced around 2014 which tried to tackle a different problem known as “Byzantine Generals Problem”. This is the problem when there are various nodes communicating with each other and a few of them are “Byzantine” which means either faulty or malicious. A good consensus protocol should be able to tackle this issue in a decentralised manner for better security of the network. One of the consensus algorithms that manage to do it is called “Federated Byzantine Agreement”. This was introduced by Stellar.
+   In FBA, every node has its “quorum set” which is a list of other nodes which this particular node trusts. Therefore to achieve consensus for a transaction, the node relies on the members in its quorum set. As long as the majority of nodes are not malicious, this will ensure security of the system. Complex web of overlapping quorum slices makes it almost impossible for the majority of nodes to collude to control consensus. FBA can be used on top of PoW or PoS (to be discussed later). Safety and fault tolerance are [preferred over liveness](https://www.youtube.com/watch?v=aU08km2xrz0&ab_channel=Lumenauts). In case of an accidental fork, the system is halted until consensus is reached. This is important in case of banking applications. Consensus that requires only message passing followed by voting process leads to high transaction volume per second and less expenditure of electricity than PoW.
+   However, a criticism of pure FBA is that it leads to fragile structures of constituent nodes, permitting topology scenarios where a single node failure can cause a network-wide failure
+   (Resistant to Forks?)
+
+   TODO: Drawback: Single point failure if quorum sets share a single node?
```

Comment:
Is Single point failure if quorum sets share a single node possible in case of Stellar FBA? I know in pure FBA this can be a problem which is why UNL with adjustable parameters are introduced in Flare.

Reply:
Yes, this is exactly what you are introducing earlier: safety and fault tolerance are preferred over liveness. When a network partition happens, maybe because the only node in common between two big sets of nodes goes down, the smaller partition will no longer be live, i.e. it won't make progress. If no partition is big enough to achieve consensus, the whole network will halt. This happened for Ripple twice in the last couple of months.